manifests/params.pp

```diff
@@ -133,27 +133,23 @@
     }
   }
 
-  if fact('aio_agent_version') =~ String[1] {
-    $puppet_ssldir = '/etc/puppetlabs/puppet/ssl'
-  } else {
-    $puppet_ssldir = '/var/lib/puppet/ssl'
-  }
+  $foreman_pki_dir = '/etc/foreman_pki'
 
   # If CA is specified, remote Foreman host will be verified in reports/ENC scripts
-  $client_ssl_ca   = "${puppet_ssldir}/certs/ca.pem"
+  $client_ssl_ca   = "${foreman_pki_dir}/ca/ca.crt"
   # Used to authenticate to Foreman, required if require_ssl_puppetmasters is enabled
-  $client_ssl_cert = "${puppet_ssldir}/certs/${lower_fqdn}.pem"
-  $client_ssl_key  = "${puppet_ssldir}/private_keys/${lower_fqdn}.pem"
+  $client_ssl_cert = "${foreman_pki_dir}/foreman/client.crt"
+  $client_ssl_key  = "${foreman_pki_dir}/foreman/client.key"
 
   $vhost_priority = '05'
 
   # Set these values if you want Passenger to serve a CA-provided cert instead of puppet's
-  $server_ssl_ca    = "${puppet_ssldir}/certs/ca.pem"
-  $server_ssl_chain = "${puppet_ssldir}/certs/ca.pem"
-  $server_ssl_cert  = "${puppet_ssldir}/certs/${lower_fqdn}.pem"
+  $server_ssl_ca = "${foreman_pki_dir}/ca/ca.crt"
+  $server_ssl_chain = "${foreman_pki_dir}/ca/ca.crt"
+  $server_ssl_cert = "${foreman_pki_dir}/apache/apache.crt"
   $server_ssl_certs_dir = '' # lint:ignore:empty_string_assignment - this must be empty since we override a default
-  $server_ssl_key   = "${puppet_ssldir}/private_keys/${lower_fqdn}.pem"
-  $server_ssl_crl   = "${puppet_ssldir}/crl.pem"
+  $server_ssl_key = "${foreman_pki_dir}/apache/apache.key"
```

Comment:
I had a question regarding that. Does the websockify key/cert have to be *same* as Apache or from the same CA as Apache's certificate?

Reply:
Ideally we'd refactor things in Foreman so websockify doesn't need to start a HTTPS server at all. That simplifies things so much.

The design would be:
* Foreman generates an encrypted JWT with the data
* Foreman sends the JWT to the client
* Client does a request on `/vnc` (or similar)
* Apache is a reverse proxy to a constantly running websockify process
* Websockify can decrypt the JWT
* Websockify works as it previously did

We can probably make websockify systemd socket activated and start lazily as an optimization, but that doesn't really change anything on the overall design.

This should also work properly in a load balanced setup as long as every webserver also runs websockify (config mgmt should ensure this) and can decrypt JWT.

I know it's a bigger refactor, but I think it's worth it for the simplification. Also for the admin since you no longer have to open ports 5910 - 5930. Security wise it means you can keep the Apache certs set to root:root with 640 which is increased security.

Something else I thought about was relying on the Apache wide certificates and not pass in any certs by default. Then the installer can configure those and leave the explicit parameters in case a user wants to override. Haven't thought this through that much, but perhaps it's easier to deal with.

Reply:
> I know it's a bigger refactor, but I think it's worth it for the simplification.

If you had to estimate the amount of work this is, in a high level fashion, what would you say? I want to balance that direction with the certificate handling work.